components/ContactList.tsx

```diff
@@ -0,0 +1,160 @@
+import { Typography } from './Typography';
+import { Separator, XStack, YStack } from 'tamagui';
+import * as Contacts from 'expo-contacts';
+import { Icon } from './Icon';
+import { useEffect, useState } from 'react';
+import { useContacts } from '@/services/contacts.service';
+import contactsRepository from '@/db/repositories/contacts.repository';
+import { Image } from 'expo-image';
+import { useQueryClient } from '@tanstack/react-query';
+import { Card } from './Card';
+import { Linking, Platform } from 'react-native';
+import { useTranslation } from 'react-i18next';
+
+export const ContactList = () => {
+  const { t } = useTranslation();
+  const queryClient = useQueryClient();
+  const [contacts, setContacts] = useState<Contacts.Contact[]>([]);
+  const { data: contactsData } = useContacts();
+
+  const handlePickContacts = async () => {
+    const { status } = await Contacts.requestPermissionsAsync();
+    if (status === 'granted') {
+      const contact = await Contacts.presentContactPickerAsync();
+      if (contact && contact.id) {
+        await createContact(contact.id);
+        queryClient.invalidateQueries({ queryKey: ['contacts'] });
+      }
+    }
+  };
+
+  const createContact = async (contactId: string) => {
+    if (!contactsData || contactsData.length === 0) {
+      await contactsRepository.createContact({
+        contactIds: [contactId],
+        createdAt: new Date().toISOString(),
+        updatedAt: new Date().toISOString(),
+        deletedAt: null,
+      });
+    } else {
+      const list = new Set([...(contactsData[0].contactIds || []), contactId]);
+      await contactsRepository.updateContact(contactsData[0].id, {
+        contactIds: Array.from(list),
+        updatedAt: new Date().toISOString(),
+      });
+    }
+  };
+
+  const getContatctDetails = async (ids: string[]) => {
```

Comment:
As fi bagat si asta tot in query si scapi si de state `contacts` si de logica de aici si de useEffectul de mai jos